src/Harald.Tests/features/Connections/Infrastructure/Persistence/FindConnectionsBySenderTypeSenderIdChannelTypeChannelIdHandlerTests.cs

```diff
@@ -0,0 +1,197 @@
+            await capabilityRepository.Add(Capability.Create(
+                Guid.NewGuid(),
+                "sheep",
+                "NotTheChannelWeWant",
+                "slackUserGroupId"
+            ));
+
+            var sut = new FindConnectionsBySenderTypeSenderIdChannelTypeChannelIdHandler(capabilityRepository);
+
+            var channelIdOfWantedChannel = new ChannelId(idOfWantedChannels);
+            var query = new FindConnectionsBySenderTypeSenderIdChannelTypeChannelId(
+                new SenderTypeCapability(),
+                null,
+                new ChannelTypeSlack(),
+                channelIdOfWantedChannel
+            );
+
+            // Act
+            var results = await sut.HandleAsync(query);
+
+            // Assert
+            Assert.All(
+                results,
+                connection =>
+                    Assert.Equal(channelIdOfWantedChannel, connection.ChannelId)
+            );
+        }
+
+        // GIVEN query with SenderId set
+        // EXPECT only capabilities with given id
+        [Fact]
+        public async Task GIVEN_query_with_SenderId_set_EXPECT_only_capabilities_with_given_id()
+        {
+            // Arrange
+            var idOfWantedSender = Guid.NewGuid();
+            var capabilityRepository = new StubCapabilityRepository();
+            await capabilityRepository.Add(Capability.Create(
+                idOfWantedSender,
+                "foo",
+                "slackChannelId",
+                "slackUserGroupId"
+            ));
+            await capabilityRepository.Add(Capability.Create(
+                idOfWantedSender,
+                "bar",
+                "slackChannelId",
+                "slackUserGroupId"
+            ));
+            await capabilityRepository.Add(Capability.Create(
+                Guid.NewGuid(),
+                "sheep",
+                "slackChannelId",
+                "slackUserGroupId"
+            ));
+
+            var sut = new FindConnectionsBySenderTypeSenderIdChannelTypeChannelIdHandler(capabilityRepository);
+
+            var senderIdOfWantedSender = new SenderId(idOfWantedSender.ToString());
+            var query = new FindConnectionsBySenderTypeSenderIdChannelTypeChannelId(
+                new SenderTypeCapability(),
+                senderIdOfWantedSender,
+                new ChannelTypeSlack(),
+                null
+            );
+
+
+            // Act
+            var results = await sut.HandleAsync(query);
+
+            // Assert
+            Assert.All(
+                results,
+                connection =>
+                    Assert.Equal(senderIdOfWantedSender, connection.SenderId)
+            );
+        }
+
+        // GIVEN empty query
+        // EXPECT connections for all capabilities
+        [Fact]
+        public async Task GIVEN_empty_query_EXPECT_connections_for_all_capabilities()
+        {
+            // Arrange
+            var capabilityRepository = new StubCapabilityRepository();
```

Comment:
Why stub this repo? Wouldnt it be clearner (less code) to inject a EF context for CapabilityEntityFrameworkRepository and use that directly? https://docs.microsoft.com/en-us/ef/core/miscellaneous/testing/in-memory

Reply:
I could save some lines here, but the CapabilityEntityFrameworkRepository is not under test. So I prefer use the simpler or less interconnected version of keeping a collection in memory in the stub.
The EF repository for Connection would come under test when we extend this feature to have its own db table and orm.

Reply:
Personally I like to test the DbContext because often it has a lot of hidden validation logic (primary centered around navigation properties and field value conversions) that can get missed in the process